geoutils/geovector.py

```diff
@@ -162,15 +165,31 @@ def create_mask(
             transform = rio.transform.from_bounds(left, bottom, right, top, width, height)
 
         # otherwise use directly rst's dimensions
+        elif isinstance(rst, gu.Raster):
+            out_shape = rst.shape
+            transform = rst.transform
+            crs = rst.crs
+            bounds = rst.bounds
         else:
-            out_shape = rst.shape  # type: ignore
-            transform = rst.transform  # type: ignore
-            crs = rst.crs  # type: ignore
+            raise ValueError("`rst` must be either a str, geoutils.Raster or None")
+
+        # Crop vector geometries to avoid issues when reprojecting
+        left, bottom, right, top = bounds  # type: ignore
+        x1, y1, x2, y2 = warp.transform_bounds(crs, self.ds.crs, left, bottom, right, top)
+        self.ds = self.ds.cx[x1:x2, y1:y2]
 
         # Reproject vector into rst CRS
         # Note: would need to check if CRS are different
         vect = self.ds.to_crs(crs)
 
+        # Create a buffer around the features
+        if not isinstance(buffer, (int, float, np.number)):
+            raise ValueError(f"`buffer` must be a number, currently set to {type(buffer)}")
+        if buffer != 0:
+            vect.geometry = [geom.buffer(buffer) for geom in vect.geometry]
```

Comment:
What's the performance/quality advantages/disadvantages to dilating/eroding the mask after rasterization?

Reply:
No idea... But @rhugonnet suggested to do it this way in #193, saying buffer was "hyper fast". 
At least it is independent of the raster size. My guess is that for very small raster, erosion/dilation might be faster, but it will be much slower for larger rasters.

Reply:
Okay!